dashboard-app/src/features/dashboard/dataManipulation/aggregatedToGraphData.ts

```diff
@@ -1,18 +1,23 @@
 import { omit } from 'ramda';
 import { Objects } from 'twine-util';
 import { AggregatedData } from './logsToAggregatedData';
-import { toUnitDuration, abbreviateIfDateString } from './util';
+import { toUnitDuration, abbreviateDateString, isDateString } from './util';
 import { DurationUnitEnum } from '../../../types';
 import Months from '../../../lib/util/months';
 import { getColourByIndex } from '../util';
+import { truncate } from 'twine-util/string';
 
 export const aggregatedToStackedGraph = (data: AggregatedData, unit: DurationUnitEnum) => {
-  const labels = Object.keys(omit(['id', 'name'], data.rows[0]));
+  const labels = Months.sortFormatted(Object.keys(omit(['id', 'name'], data.rows[0])));
+
   return {
-    labels: labels.map((x) => abbreviateIfDateString(Months.format.abreviated, x).split(' ')),
+    labels: labels.map((x) =>
+      isDateString(x)
```

Comment:
What's the purpose of this ternary? From reading through it is unclear why the string is truncated if it is not a valid date. Can we wrap this in a named function that describes its purpose

Reply:
This is looking ahead to the projects charts, where the x-axis labels are activities (and hence very long). Some labels will be hidden if they don't all fit on the x-axis.

Basically we want to ensure all labels are short enough so that all labels are displayed.

This could probably be made more simple by just doing `truncate(abbreviateIfDateString(Months.format.abbreviated), 10)` since we know the month strings will always be 8 characters long and thus wont be truncated.

Reply:
Yeah, that works. Or wrap it in a function called `abbreviateLables`, that internally abbreviates date strings and truncates everything else to 10 characters